tests/integration/test_gomod_packages.py

```diff
@@ -32,3 +34,34 @@ def test_gomod_vendor_without_flag(test_env):
         )
     else:
         utils.assert_properly_completed_response(completed_response)
+
+
+@pytest.mark.parametrize("env_name", [("wrong_vendor"), ("empty_vendor")])
+def test_gomod_vendor_check_fail(env_name, test_env):
+    """
+    Validate failing of gomod vendor request with gomod-vendor-check flag and inconsistent vendor.
+
+    Checks:
+    * The request fails with expected error message
+    """
+    env_data = utils.load_test_data("gomod_vendor_check.yaml")[env_name]
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"], test_env.get("timeout"))
+    initial_response = client.create_new_request(
+        payload={
+            "repo": env_data["repo"],
+            "ref": env_data["ref"],
+            "flags": env_data["flags"],
+            "pkg_managers": env_data["pkg_managers"],
+        },
+    )
+    completed_response = client.wait_for_complete_request(initial_response)
+    assert completed_response.status == 200
+    assert completed_response.data["state"] == "failed"
+    error_msg = (
+        "The content of the vendor directory is not consistent with go.mod. "
+        "Run `go mod vendor` locally to fix this problem. See the logs for more details."
```

Comment:
It might also be interesting to check the logs here, since we promise there will be more details :D

AFAIK it's not something we've done in integration tests yet so I'd be OK with leaving it as a future improvement, I'll leave it up to you.

Reply:
Were you thinking that we confirm the details of the log (the same way we confirm the `error_msg`) in the test function as well? The logs we are looking for are logged in `_vendor_changed` - how can I access the logs in `test_gomod_vendor_check_fail`?

Reply:
Yeah that's what I was thinking. Request logs can be accessed at a url, e.g. localhost:8080/api/v1/requests/1/logs